common/socket/src/main/java/io/helidon/common/socket/IdleInputStream.java

```diff
@@ -107,20 +122,46 @@ boolean isClosed() {
 
     private void handle() {
         try {
-            next = upstream.read();
-            if (next <= 0) {
-                closed = true;
+            //Currently configured socket read timeout. This is intended to be restored after this method finishes
+            int toRestore = -1;
+            for (int i = 0; !cancelled; i++) {
+                try {
+                    toRestore = timeoutToRestore(toRestore);
+                    socket.setSoTimeout(TIMEOUT);
+                    next = upstream.read();
+                    if (next <= 0) {
+                        closed = true;
+                        return;
+                    }
+                    break;
+                } catch (SocketTimeoutException e) {
+                    if (i + 1 >= idleTimeoutIterations) {
```

Comment:
Do we need to throw the last socket timeout ?
Ins't it better to ignore it them and let the consuming code do another read ?

Reply:
I am trying to have the same functionality as we had before and we were throwing the exception there.